### PR TITLE
MM-T639 Webhook DM test

### PR DESCRIPTION
**Pull request by mvitale1989**
#### Summary

This PR adds an E2E test case, to verify that an Incoming Webhook can successfully create a DM to another user.

#### Ticket Link

Fixes https://github.com/mattermost/mattermost-server/issues/18183

#### Related Pull Requests
NONE

#### Screenshots
No UI change

#### Release Note
```release-note
NONE
```

### CODE REVIEW

**Comment by furqanmlk** [comment]
@mvitale1989 can you please fix lint?

**Comment by mvitale1989** [comment]
Sure, now it's fixed

**Review by furqanmlk — APPROVED** [review]
LGTM
image

File: e2e/cypress/tests/integration/integrations/incoming_webhook/incoming_webhook_creates_dm_spec.js [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+            should('be.visible').
+            get('.col__name > button.user-popover').
+            contains(incomingWebhookConfiguration.display_name);
+        cy.getLastPost().
+            should('be.visible').
+            get('.col__name .BotBadge > span').
+            contains('BOT');
+        cy.getLastPost().
+            get('.post-message__text').
+            contains(incomingWebhookText);
```
Comment (Phrynobatrachus) [inline e2e/cypress/tests/integration/integrations/incoming_webhook/incoming_webhook_creates_dm_spec.js:69]:
```suggestion
       // # Verify that the channel was created correctly with an unread message, and open it
        cy.uiGetLHS().
            contains(`${generatedUser.username}`).
            should('have.class', 'unread-title').
            click();

        cy.getLastPost().within(($post) => {
            cy.wrap($post).contains(incomingWebhookConfiguration.display_name).should('be.visible');
            cy.wrap($post).contains('BOT').should('be.visible');
            cy.wrap($post).contains(incomingWebhookText).should('be.visible');
        });
```

Keeps the selections a touch neater.

**Review by Phrynobatrachus — CHANGES_REQUESTED** [review]
Hello, thank you for the PR! I have a small suggestion for the UI portion, but the setup code and overall test looks solid.

File: e2e/cypress/tests/integration/integrations/incoming_webhook/incoming_webhook_creates_dm_spec.js (hunk 1 above)
Comment (mvitale1989) [reply e2e/cypress/tests/integration/integrations/incoming_webhook/incoming_webhook_creates_dm_spec.js:69]: Hello, thanks for the suggestion! Will add it as a local commit, to fix the indentation of the first comment line as well

**Review by Phrynobatrachus — APPROVED** [review]

**Pull request merged**